### ISSUE TITLE
stability: never terminating "txn coordinator" request

### ISSUE THREAD

**Issue by petermattis** [cockroachdb/cockroach#11668]
Perhaps related to #11667. Seen on `delta/104.196.175.87`:

```
2016/11/28 16:22:42.440774	16826.904537	txn coordinator
16:22:42.440859	 .    85	... [n4] querying next range at /Table/51/1/4453809139528403303/"027256f6-5ddd-4eb3-8ccd-dbecfa9c6b94"/48011/0
16:22:42.440867	 .     8	... [n4] meta descriptor lookup
16:22:42.440883	 .    16	... [n4] looked up range descriptor from cache
16:22:42.528837	 . 87954	... [n4] sending RPC for batch: 1 Gc
16:22:43.032381	 .503544	... [n4] timeout, trying next peer
16:22:43.032689	 .   308	... [n4] error, trying next peer: range 162: replica {4 7 2} not lease holder; node_id:7 store_id:13 replica_id:3 is
```

`txn coordinator` is the span started by `TxnCoordSender.Send`. Seems worrisome that this has never completed. I see one goroutine that might be related:

```
goroutine 27042124 [select, 284 minutes]:
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*Stream).Header(0xc44065e690, 0x1b4ed98, 0xc42ad51b90, 0x2529f80)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/transport.go:239 +0x299
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.recvResponse(0x25223e0, 0x2a30ce8, 0x0, 0x0, 0x0, 0x0, 0x2513b60, 0xc42120adc0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/call.go:62 +0xad
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.Invoke(0x25241a0, 0xc449978f40, 0x1995e3b, 0x21, 0x1956440, 0xc448180c08, 0x18dc8e0, 0xc43f53f140, 0xc42046e900, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/call.go:202 +0x8a1
github.com/cockroachdb/cockroach/pkg/roachpb.(*internalClient).Batch(0xc420123ca8, 0x25241a0, 0xc449978f40, 0xc448180c08, 0x0, 0x0, 0x0, 0x25241a0, 0xc4264bfe80, 0x25241a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:1499 +0xd2
github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func2(0xc448180be0, 0xc4312703c0, 0xc43f53ed20)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:193 +0x94
created by github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:203 +0x364
```

I'm also seeing a number of goroutines stuck trying to acquire leases:

```
261: select [0~46 minutes] [Created by grpc.(*Server).serveStreams.func1 @ server.go:423]
    storage             replica.go:931              (*Replica).redirectOnOrAcquireLease(*, #86, *, *)
    storage             replica.go:1820             (*Replica).tryAddWriteCmd(*, #86, *, *, 0, *, *, *, 0, *, ...)
    storage             replica.go:1715             (*Replica).addWriteCmd(*, #86, *, *, 0, *, *, *, 0, *, ...)
    storage             replica.go:1183             (*Replica).Send(*, #86, *, *, 0, *, *, *, 0, *, ...)
    storage             store.go:2386               (*Store).Send(*, #86, *, *, 0, *, *, *, 0, *, ...)
    storage             stores.go:187               (*Stores).Send(#326, #86, *, 0, 0, *, *, *, 0, *, ...)
    server              node.go:819                 (*Node).batchInternal.func1(0, 0)
    stop                stopper.go:245              (*Stopper).RunTaskWithErr(#313, *, 0, 0)
    server              node.go:830                 (*Node).batchInternal(#222, #86, *, *, *, *, *)
    server              node.go:850                 (*Node).Batch(#222, #86, *, *, *, *, *)
    roachpb             api.pb.go:1522              _Internal_Batch_Handler(#34, #222, #86, *, *, 0, 0, 0, *, *)
    grpc                server.go:611               (*Server).processUnaryRPC(#396, #95, *, *, #528, #59, 0, 0, 0)
    grpc                server.go:769               (*Server).handleStream(#396, #95, *, *, 0)
    grpc                server.go:422               (*Server).serveStreams.func1.1(*, #396, #95, *, *)
```

**Comment by petermattis** [comment]
This still seems to be happening on `delta`. From moments ago:

```
2016/11/30 21:49:12.012896	2732.101004	txn coordinator
21:49:12.013002	 .   106	... [n8] querying next range at /Table/51/1/5857242849329754856/"ba382684-3983-446f-a4e5-73ec658ee4fb"/21331/0
21:49:12.013011	 .     9	... [n8] meta descriptor lookup
21:49:12.013025	 .    14	... [n8] looked up range descriptor from cache
21:49:12.013073	 .    47	... [n8] sending RPC for batch: 1 Gc
21:49:12.513824	 .500751	... [n8] timeout, trying next peer
21:49:12.515875	 .  2052	... [n8] error, trying next peer: range 27: replica {4 8 8} not lease holder; node_id:8 store_id:16 replica_id:3 is
```

**Comment by petermattis** [comment]
Cc @cockroachdb/stability, @bdarnell

**Comment by petermattis** [comment]
Wiped and restarted `delta` earlier today and so far this problem hasn't reappeared.

**Comment by petermattis** [comment]
I'm going to claim this was fixed by #12000 and will re-open if it reappears.
